### PR TITLE
Explanation purpose for using 16x8 MLIR quantizer

### PR DESCRIPTION
**Pull request by tucan9389**
Since there was no python level interface to use the MLIR quantizer for int16x8, I made a quick example.

Context: https://github.com/tensorflow/tensorflow/issues/59626

### CODE REVIEW

**Comment by blaine-fs** [comment]
@tucan9389  thanks a lot for the help. Am I correct in interpreting this PR as modifying TF to enable the MLIR quantizer for 16x8 ops? Are there any plans to merge this into `master`?

**Comment by tucan9389** [comment]
@blaine-fs 
Thanks for asking. This PR won't be merged into master at this moment.
Ops btw the `lite.py` in this PR is only the changes to use MLIR quantizer, but there are other changes. I'll fix it soon.

**Comment by tucan9389** [comment]
@blaine-fs 

> Am I correct in interpreting this PR as modifying TF to enable the MLIR quantizer for 16x8 ops? 

Yes, it's correct. This PR is for only 16x8 LSTM model. 
We have two ways to convert to 16x8.
1) With this PR, convert to 16x8 by using MLIR quantizer, and then use the TFLM requantizer(16x8(b:32)→16x8(b:64))
2) Without this PR, convert to 8x8 and use the requantizer(8x8(a:8, w:8, b:32)→16x8(a:16, w:8, b:64))

If you go with 1), 16x8 MLIR quantizer supports only unidirectional LSTM op, softmax op, fc op. 
On the other hand, with 2), 8x8 MLIR quantizer supports a lot of ops. But I think the 1) would be a less accuracy loss than 2).

**Comment by blaine-fs** [comment]
Hi @tucan9389 ,

I rebuilt TF using only this PR, and I was able to get this [IMDB sample](https://github.com/femtosense/external_imdb_lstm_sample) compiling in int16 mode! I saw a few % accuracy bump over the 8x8 version.

Thank you very much for the example! This is a huge help to my organization internally, and I am also hoping we can use in MLPerf Tiny it to develop a noise cancellation benchmark.

Regarding your options above, is the purpose of the requantizer to convert 32-bit biases to 64-bit? Do you find that has a large impact on model accuracy? My organization is running TFLite on a custom HW accelerator which does not support 64-bit biases anyways, so I think we are fine with (w: 8, a: 16, b: 32) as output by the 16x8 MLIR quantizer.

**Comment by blaine-fs** [comment]
> @blaine-fs Thanks for asking. This PR won't be merged into master at this moment. Ops btw the `lite.py` in this PR is only the changes to use MLIR quantizer, but there are other changes. I'll fix it soon.

I'm sorry to hear that. Is that because this change breaks some existing tests? Is there any way we could help you merge this into master? Or could it be gated by a TFLite converter flag?
